dispatch/tests/test_api_pages.py

```diff
@@ -0,0 +1,226 @@
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # Check Data
+        self.assertEqual(response.data['title'], 'Test Page')
+        self.assertEqual(response.data['slug'], 'test_page')
+        self.assertEqual(response.data['snippet'], "This is a test snippet")
+        self.assertEqual(response.data['content'][0]['type'], 'paragraph')
+        self.assertEqual(response.data['content'][0]['data'], 'This is some paragraph text')
+
+    def test_update_fields_page(self):
+        """
+        Should be able to update the title, slug, snippet of the page
+        """
+
+        # First make a page
+        page = self._create_page()
+
+        # Generate detail url
+        url = reverse('api-pages-detail', args=[page.data['id']])
+
+        # Change the fields
+        new_data = {
+          "title": "New Test Page",
+          "slug": "New test-page",
+          "snippet": "This is a new test snippet",
+          "content": [
+            {
+              "type": "paragraph",
+              "data": "This is some paragraph text"
+            }
+          ]
+        }
+
+        response = self.client.patch(url, new_data, format='json')
+
+        # Check data
+        self.assertEqual(response.data['title'], "New Test Page")
+        self.assertEqual(response.data['slug'], "New test-page")
+        self.assertEqual(response.data['snippet'], "This is a new test snippet")
+
+    def test_update_content_page(self):
+        """
+        Should be able to update contents of the page
+        """
+
+        # First make a page
+        page = self._create_page()
+
+        # Generate detail url
+        url = reverse('api-pages-detail', args=[page.data['id']])
+
+        # Change the content
+        new_data = {
+          "content": [
+            {
+              "type": "paragraph",
+              "data": "This is some brand new paragraph text"
+            }
+          ]
+        }
+
+        response = self.client.patch(url, new_data, format='json')
+
+        self.assertEqual(response.data['content'][0]['data'], "This is some brand new paragraph text")
+
+    def test_update_content_unauthorized(self):
+        """
+        Patch page should fail with unauthenticated request
+        """
+
+        # First make a page
+        page = self._create_page()
+
+        # Generate detail url
+        url = reverse('api-pages-detail', args=[page.data['id']])
+
+        self.client.credentials()
+
+        # Change the fields
+        new_data = {
+          "content": [
+            {
+              "type": "paragraph",
+              "data": "This is some new paragraph text"
+            }
+          ]
+        }
+
+        response = self.client.patch(url, new_data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        # Check Data
+        self.assertEqual(response.data['content'][0]['data'], 'This is some paragraph text')
```

Comment:
An unauthorized request actually won't return any data, so this line can be removed